### ISSUE TITLE
If Persona's login dialog is blocked by the popup blocker and the popup is allowed, it is blank

### ISSUE THREAD

**Issue by mattbasta** [mozilla/persona#4055]
STR:
1. Trigger Persona's `navigator.id.request()` from something other than a user action
2. The popup blocker will kick in
3. Manually allow the popup

You'll see the login dialog load, but there is no text, only the big arrow in the background.

**Comment by callahad** [comment]
Triaging as 1-star. Calling `request()` without user action is not intended to work.

Leaning toward wontfix -- @fmarier @seanmonstar, care to weigh in?

**Comment by mattbasta** [comment]
In this instance, it's not that the user did not choose to open the popup,
it's that persona is being triggered inside an iframe via postmessage,
which does not carry user intent.

**Comment by lukehutch** [comment]
I'm having an issue with this -- I'm trying to use Polymer's declarative event handler binding so that I can enforce CSP on my website during Persona login. I'm doing so as follows:

``` html
<polymer-element name="persona-auth">
  <template>
    <button id="signInBtn" hidden on-tap="{{onSignIn}}">Sign in</button>
    <button id="signOutBtn" hidden on-tap="{{onSignOut}}">Sign out</button>
  </template>
  <script>
    var currentUser = $.cookie('_email');
    if (currentUser === undefined) {
      currentUser = null;
    }

    Polymer({
      onSignIn: function() {
        navigator.id.request();
      },
      onSignOut: function() {
        navigator.id.logout();
      },
      ready: function() {
        if (currentUser === null) {
            this.$.signInBtn.removeAttribute("hidden");
        } else {
            this.$.signOutBtn.removeAttribute("hidden");            
        }
      }
    });

    navigator.id.watch({
      loggedInUser: currentUser,
      onlogin: function(assertion) {
        $.ajax({
          type: 'POST',
          url: '/persona-login',
          data: {assertion: assertion},
          success: function(res, status, xhr) {
            window.location.reload();
          },
          error: function(xhr, status, err) {
            console.log('Login failure: ' + err);
            $.removeCookie('_email');
            window.location.reload();
          }
        });
      },
      onlogout: function() {
        $.ajax({
          type: 'POST',
          url: '/persona-logout',
          success: function(res, status, xhr) {
            console.log(res + " " + status + " " + xhr)
            window.location.reload();
          },
          error: function(xhr, status, err) {
            console.log('Logout failure: ' + err);
          }
        });
      }
    });
  </script>
</polymer-element>
```

I get the same effect described by mattbasta: the popup is blocked, and if I manually show the popup, I just see an empty Persona popup that only has the big arrow icon and nothing else.

Am I missing something? Is there a correct way to wire up the sign in and sign out events declaratively? I hope Persona can be made to work with CSP and with Polymer.

**Comment by callahad** [comment]
Hi! To help us better focus, I'm "closing" all issues that have been open for more than six months. These have been tagged "cleanup-2014" so that we can go back and review them in the future.

For more information, check out this thread: http://thread.gmane.org/gmane.comp.mozilla.identity.devel/7394

If you believe this bug is still a major issue for you, please comment, submit a pull request, or discuss it on our mailing list: https://lists.mozilla.org/listinfo/dev-identity

Sorry for GitHub notification spam!

**Comment by lukehutch** [comment]
I think this bug was accidentally closed by a runaway script like the other two I filed... ping :-)
